src/odemis/driver/xt_client.py

```diff
@@ -2179,16 +2205,42 @@ def __init__(self, name: str, role: str, parent: model.HwComponent, rng: dict =
                                                 readonly=True)
         self._updatePosition()
 
+        # update the offset if raw stage coordinates are read
+        self._get_offset_raw_coordinate_system()
```

Comment:
This function currently does actually 2 things: measure the offset, and change the raw coordinates. The name implies it just does one (and return the value, which it doesn't do).
I'd suggest to call it like `self._switch_coordinate_system(self._raw_coordinates)`. (it takes 1 argument)